fitbit-react/src/Pages/ProductsPage/ProductListing/ProductListing.js

```diff
@@ -1,26 +1,91 @@
 import React from "react";
-import { VerticalCard, products } from "../index";
+import {
+  VerticalCard,
+  products,
+  useFilter,
+  filterByCategory,
+  sortByHighToLow,
+  sortByLowToHigh,
+} from "../index";
 
 function ProductListing() {
+  const { filters, dispatch } = useFilter();
+
+  const getSortedData = (state, filteredData) => {
+    let updatedData = [...filteredData];
+    if (state.sortBy === "highToLow") {
+      sortByHighToLow(updatedData);
+    } else if (state.sortBy === "lowToHigh") {
+      sortByLowToHigh(updatedData);
+    }
+    return updatedData;
+  };
+  const getFilteredData = (filters) => {
+    let updatedData = [...products];
+    let equipmentCategory = [];
+    let clothesCategory = [];
+    let glovesCategory = [];
+    let dumbellsCategory = [];
+
+    if (filters.categories.equipments) {
+      equipmentCategory = filterByCategory(updatedData, "equipment");
+    }
+    if (filters.categories.clothes) {
+      clothesCategory = filterByCategory(updatedData, "clothes");
+    }
+    if (filters.categories.handGloves) {
+      glovesCategory = filterByCategory(updatedData, "gloves");
+    }
+    if (filters.categories.dumbells) {
+      dumbellsCategory = filterByCategory(updatedData, "dumbells");
+    }
+    updatedData = [
+      ...equipmentCategory,
+      ...clothesCategory,
+      ...glovesCategory,
+      ...dumbellsCategory,
+    ];
+    if (!filters.includeOutOfStock) {
+      updatedData = updatedData.filter((data) => data.inStock === true);
+    }
+    if (filters.fastDelivery) {
+      updatedData = updatedData.filter((data) => data.fastDelivery === true);
+    }
+    updatedData = updatedData.filter(
+      (data) => parseInt(data.rating, 10) >= parseInt(filters.rating, 10)
+    );
+    return updatedData;
+  };
+  let filteredData = getFilteredData(filters);
+  let sortedData = getSortedData(filters, filteredData);
   return (
     <div className="products-container">
       <div className="spacer-1"></div>
-      {products?.map(
+      {sortedData?.map(
         ({
           productId,
           productTitle,
           discountedPrice,
           discountPercent,
           realPrice,
           productImageUrl,
+          category,
+          rating,
+          inStock,
+          fastDelivery,
         }) => (
           <VerticalCard
             key={productId}
+            productId={productId}
             productTitle={productTitle}
             discountedPrice={discountedPrice}
             realPrice={realPrice}
             discountPercent={discountPercent}
             productImageUrl={productImageUrl}
+            category={category}
+            inStock={inStock}
+            rating={rating}
+            fastDelivery={fastDelivery}
```

Comment:
Hmmm, just a suggestion, if you are JUST passing these values to next component, then rather than destructuring 1 by 1 and then passing them 1 by 1, you can directly pass and destructure where you really want to use them.

Reply:
Ok i will consider this suggestion, if i will not figure it out i will take your help in this